jetstream/cli.py

```diff
@@ -1045,3 +1096,250 @@ def ensure_enrollments(
+    # given, find it from the config file.
+    if not experiment_slug:
+        external_config = entity_from_path(Path(config_file))
+        experiment_slug = [external_config.slug]
+
+    for slug in experiment_slug:
+        click.echo(f"Generate preview for {slug}")
+        table = bq_normalize_name(slug)
+
+        # delete previously created preview tables if exist
+        client = BigQueryClient(project_id, dataset_id)
+        client.delete_experiment_tables(slug, analysis_periods, delete_enrollments=True)
+
+        collection = ExperimentCollection.from_experimenter()
+        experimenter_experiments = collection.with_slug(slug)
+        config_getter = ConfigLoader.with_configs_from(config_repos).with_configs_from(
+            private_config_repos, is_private=True
+        )
+
+        experiment: Optional[Experiment] = None
+        if experimenter_experiments.experiments != []:
+            experiment = experimenter_experiments.experiments[0]
+
+        # set dummy experiment values and adjust dates
+        experiment = Experiment(
+            experimenter_slug=experiment.experimenter_slug if experiment else slug,
+            normandy_slug=experiment.normandy_slug if experiment else slug,
+            type=experiment.type if experiment else "v6",
+            status="Live",
+            start_date=start_date - timedelta(days=3),  # subtract enrollment days
+            end_date=end_date,
+            proposed_enrollment=enrollment_period,
+            branches=experiment.branches if experiment else [Branch(slug="control", ratio=1)],
+            reference_branch=experiment.reference_branch if experiment else "control",
+            is_high_population=False,
+            app_name=platform,
+            app_id=PLATFORM_CONFIGS[platform].app_id,
+            outcomes=experiment.outcomes if experiment else [],
+            enrollment_end_date=None,
+        )
+
+        spec = AnalysisSpec.default_for_experiment(experiment, config_getter.configs)
+        if external_spec := config_getter.spec_for_experiment(experiment.normandy_slug):
+            spec.merge(external_spec)
+
+        config = spec.resolve(experiment=experiment, configs=config_getter.configs)
+
+        # generated sampled enrollment query
+        if generate_population:
+            spec.experiment.enrollment_query = sampled_enrollment_query(
+                start_date, config, population_sample_size
+            )
+
+            # update dates
+            spec.experiment.start_date = (start_date - timedelta(days=3)).strftime("%Y-%m-%d")
+            spec.experiment.end_date = end_date.strftime("%Y-%m-%d")
+            spec.experiment.enrollment_period = enrollment_period
+
+            # set experiments_column_type to none for all data sources used in the experiment;
+            # this needs to be done otherwise all clients will be filtered out since none
+            # are enrolled in the experiment
```

Comment:
For experiments that actually exist specifying `--generate-population` wouldn't be strictly necessary. If `--generate-population` is not specified then it would just run the default enrollment query.